app/flatpak-builtins-install.c

```diff
@@ -521,16 +542,29 @@ flatpak_builtin_install (int argc, char **argv, GCancellable *cancellable, GErro
                                           &matched_kinds, &id, &arch, &branch, error))
         return FALSE;
 
-      ref = flatpak_dir_find_remote_ref (dir, remote, id, branch, default_branch, arch,
-                                         matched_kinds, &kind, cancellable, error);
+
+      if (opt_no_pull)
+        {
+          if (!flatpak_resolve_single_kind (matched_kinds, &kind, error))
+            return FALSE;
+
+          ref = flatpak_compose_ref (kind == FLATPAK_KINDS_APP, id, branch, arch, error);
```

Comment:
This seems a bit simple, why not have a copy of flatpak_dir_find_remote_ref() that works on refs in the local repo instead? It would also be similar to flatpak_dir_find_installed_refs(), but work on the repo, not the deployed dirs.

Reply:
I suppose I could look into that, yeah

Reply:
Okay, I implemented this by using `ostree_repo_list_refs` and factored out a little helper function